hudi-common/pom.xml

```diff
@@ -154,6 +155,7 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-hdfs</artifactId>
+      <scope>provided</scope>
```

Comment:
wondering if the parent pom scope is already `provided` , is nt that inherited>?

Reply:
+1, I think we can remove all the unnecessary Hadoop dependency from the sub-modules since they already exist in the parent pom. Do you mind removing those in this PR @ZhangChaoming ?

Reply:
@garyli1019 My pleasure.